repository_service_tuf_worker/repository.py

```diff
@@ -1018,6 +1104,67 @@ def _update_metadata_delegation(
 
         return (success, failed)
 
+    def _update_delegated_role_target_files(self, delegation, db_role):
+        delegation.signed.targets.clear()
+        delegation.signed.targets = {
+            file.path: TargetFile.from_dict(file.info, file.path)
+            for file in db_role.target_files
+            if file.action == targets_schema.TargetAction.ADD
+            # Filtering the files with action 'ADD' cannot be done
+            # in CRUD. If a target role doesn't have any target
+            # files with an action 'ADD' (only 'REMOVE') then using
+            # CRUD will not return the target role and it won't be
+            # updated. An example can be when there is a role with
+            # one target file with action "REMOVE" and the CRUD
+            # will return None for this specific role.
+        }
+
+        return delegation
+
+    def update_targets_delegated_role(self, role: str):
+
+        if role == Targets.type:
+            targets: Metadata[Targets] = self._storage_backend.get(
+                Targets.type
+            )
+            self._bump_and_persist(targets, Targets.type)
+
+            return {
+                Targets.type: {
+                    "version": targets.signed.version,
+                    "expire": targets.signed.expires,
+                    "target_files": [],
+                }
+            }
+
+        signer = self._signer_store.get(self._online_key)
+        expire = self._settings.get_fresh("BINS_EXPIRATION")
+
+        db_target_role: targets_crud.models.RSTUFTargetRoles = (
+            targets_crud.read_role_joint_files(self._db, role)
+        )
+        rolename = db_target_role.rolename
+
+        delegation: Metadata[Targets] = self._storage_backend.get(
+            rolename, db_target_role.version
+        )
+        self._update_delegated_role_target_files(delegation, db_target_role)
+        role_target_files = [file.path for file in db_target_role.target_files]
+
+        # Update expiry, bump version, and persist
+        self._bump_and_persist(
+            delegation, BINS, persist=False, signer=signer, expire=expire
+        )
```

Comment:
How do you know that the person is using `BINS` and not custom delegation?

Reply:
This is not covered here.
I'll work to handle the case of custom delegations, not sure if already in this PR.